Fix wrong behavior of is_alias_of and c10d::reducer on MTIA

egienvalue opened the pull request:
Reviewed By: kirteshpatil

Differential Revision: D51860023




cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

fduwjj commented on torch/csrc/distributed/c10d/reducer.cpp:
```diff
@@ -1257,18 +1280,27 @@ void Reducer::populate_bucket_views_out(
-      // Fall back to a C-style contiguous view, again anticipating
-      // AccumulateGrad will do the same when stashing grads for non-dense
-      // params.
+    if (v.is_mtia()) {
```
Can we move this into else condition?

fduwjj commented on torch/csrc/distributed/c10d/reducer.cpp:
```diff
@@ -1206,18 +1220,27 @@ void Reducer::initialize_bucket_views(Reducer::Bucket& bucket) {
-      // Fall back to a C-style contiguous view, again anticipating
-      // AccumulateGrad will do the same when stashing grads for non-dense
-      // params.
+    if (v.is_mtia()) {
```
ditto.

fduwjj reviewed (approved):
LGTM

egienvalue committed:
Fix wrong behavior of is_alias_of and c10d::reducer on MTIA (#115553)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/115553

Reviewed By: kirteshpatil, fduwjj

Differential Revision: D51860023
